.github/workflows/lint.yml

```diff
@@ -47,8 +35,10 @@ jobs:
     - name: prepare workspace
       continue-on-error: true
       run: make prepare_local_workspace
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     - name: golangci-lint
       uses: golangci/golangci-lint-action@55c2c1448f86e01eaae002a5a3a9624417608d84 # v6
       with:
-        version: v1.64.6
+        version: v2.4.0
```

Comment:
@kangasta was this a manual change?

For your information, `pulumi/ci-mgmt` is still locked on `golangci-lint` v1.

We have an open issue to track upgrading to v2: https://github.com/pulumi/ci-mgmt/issues/1543